webserver/webserver/src/main/java/io/helidon/webserver/WebServer.java

```diff
@@ -143,6 +143,23 @@ default int port() {
      */
     int port(String socketName);
 
+    /**
+     * Update the TLS configuration of the default socket {@link WebServer#DEFAULT_SOCKET_NAME}.
+     *
+     * @param tls new TLS configuration
+     * @throws IllegalStateException if the new tls is not enabled or currently set tls is null
+     */
+    void updateTls(WebServerTls tls);
+
+    /**
+     * Update the TLS configuration of the named socket.
+     *
+     * @param tls new TLS configuration
+     * @param socketName specific named socket name
+     * @throws IllegalStateException if the new tls is not enabled or currently set tls is null
```

Comment:
Looking at the constructor for [SockerConfig](https://github.com/oracle/helidon/blob/b7cbd81c10b2fada78399f655ae1980c5ce6b8d1/webserver/webserver/src/main/java/io/helidon/webserver/ServerBasicConfig.java#L211), we don't keep the `WebServerTls` so it's not symetrical :(

How about `if {@link WebServerTls#enabled()} returns {@code false} or if {@link SocketConfiguration#ssl()} returns {@code null}.

We should probably update the ssl configuration for 3.0 @tomas-langer thoughts ?